docs/user/upgrade.rst

```diff
@@ -3,10 +3,78 @@
 Upgrade Guide
 =============
 
-This section describes code and configuration changes required when upgrading GeoMesa versions.
+This section contains general information on version upgrades, as well as version-specific changes that may
+affect the end user.
 
-Version 2.0.0
-+++++++++++++
+Compatibility Across Versions
++++++++++++++++++++++++++++++
+
+Semantic Versioning
+-------------------
+
+Starting with 2.0.0, GeoMesa is trying to adhere to `semantic versioning <https://semver.org/>`__. Essentially,
+releases are broken down into major, minor and patch versions. For a version number like 2.0.1, 2 is the major
+version, 2.0 is the minor version, and 2.0.1 is the patch version.
```

Comment:
This is phrased differently than I expected.  I kinda like it...  I was thinking that '0' was the minor version, but that's obviously relative to the Major version 2...

Reply:
yeah, i thought it was clearer that way